### ISSUE TITLE
Wrong alignment length in MSF header

### ISSUE THREAD

**Issue by peterjc** [ANHIG/IMGTHLA#201]
Consider branch 3320, aka v3.32.0, file ``msf/MICA_prot.msf`` which begins as follows:

```
$ head MICA_prot.msf
!!AA_MULTIPLE_ALIGNMENT

   MSF: 333  Type: P  Apr 17, 2018  15:22  Check: 0 ..

 Name: MICA*001         Len:   389  Check: 1197  Weight:  1.00
 Name: MICA*002:01      Len:   389  Check: 2273  Weight:  1.00
 Name: MICA*002:02      Len:   389  Check: 7220  Weight:  1.00
 Name: MICA*002:03      Len:   389  Check: 1125  Weight:  1.00
 Name: MICA*002:04      Len:   389  Check: 1125  Weight:  1.00
 Name: MICA*004         Len:   389  Check: 2139  Weight:  1.00
```

The header line says there should be 333 columns, yet most of the sequences say 389 - and indeed adding trailing padding the shorter sequences, the data might be interpreted as an alignment with 107 rows and 389 columns.

This might be related to the same root cause as #200, which is sequences shorter than the stated alignment length, whereas here we have sequences longer than the stated alignment length.

Update: See https://github.com/ANHIG/IMGTHLA/blob/3320/msf/MICA_prot.msf

**Comment by peterjc** [comment]
Another example, again branch 3320, file ``msf/DOA_prot.msf``

```
$ head -n 17 DOA_prot.msf
!!AA_MULTIPLE_ALIGNMENT

   MSF: 62  Type: P  Apr 17, 2018  15:21  Check: 0 ..

 Name: DOA*01:01:01     Len:   250  Check: 1755  Weight:  1.00
 Name: DOA*01:01:02:01  Len:   250  Check: 1755  Weight:  1.00
 Name: DOA*01:01:02:02  Len:   250  Check: 1755  Weight:  1.00
 Name: DOA*01:01:02:03  Len:   250  Check: 1755  Weight:  1.00
 Name: DOA*01:01:03     Len:   250  Check:  105  Weight:  1.00
 Name: DOA*01:01:04:01  Len:   250  Check: 1755  Weight:  1.00
 Name: DOA*01:01:04:02  Len:   250  Check:  105  Weight:  1.00
 Name: DOA*01:01:05     Len:   250  Check: 1755  Weight:  1.00
 Name: DOA*01:01:06     Len:   250  Check: 1755  Weight:  1.00
 Name: DOA*01:02        Len:   250  Check: 1035  Weight:  1.00
 Name: DOA*01:03        Len:   250  Check: 2175  Weight:  1.00
 Name: DOA*01:04N       Len:    62  Check: 8719  Weight:  1.00
//
```

Again you might change the header to say 250 columns, and then fix the truncation of ``DOA*01:04N`` (as in #200), giving a 12 row, 250 column alignment.

https://github.com/ANHIG/IMGTHLA/blob/3320/msf/DOA_prot.msf

**Comment by peterjc** [comment]
These two and one more are broken on branch 3310, aka v3.31.0

https://github.com/ANHIG/IMGTHLA/blob/3310/msf/MICA_prot.msf

https://github.com/ANHIG/IMGTHLA/blob/3310/msf/DOA_prot.msf

https://github.com/ANHIG/IMGTHLA/blob/3310/msf/G_prot.msf

**Comment by peterjc** [comment]
The first two are also broken on branch 3300 aka v3.30.0,

https://github.com/ANHIG/IMGTHLA/blob/3300/msf/MICA_prot.msf

https://github.com/ANHIG/IMGTHLA/blob/3300/msf/DOA_prot.msf

**Comment by jrob119** [comment]
For the MSF files, the sequence length varies for a number of reasons, this may be due to premature stop codons, unsequenced regions or deletions. Traditionally these were not been padded as this was not necessary for the program ReadSeq which was used to generate the MSF files. This is the reason for the varying lengths in earlier releases. Depending on the use of the MSF file, this was not causing an issues for our users. Later releases should show additionally padding to provide standardised lengths in the MSF entries, following a request. We need to look at the scope of the issue, and the need for the earlier files to be regenerated, as most sequences are available in later releases.

**Comment by peterjc** [comment]
That makes sense, I suspected a pipeline issue had been fixed as the releases since v3.32.0 (branch 3320) seem fine as regards the MSF header length.
